src/Http/Requests/DestroyThread.php

```diff
@@ -24,17 +26,38 @@ public function fulfill()
     {
         $thread = $this->route('thread');
 
-        if (config('forum.general.soft_deletes') && isset($this->validated()['permadelete']) && $this->validated()['permadelete'] && method_exists($thread, 'forceDelete'))
+        $thread->readers()->detach();
+        
+        $threadIsTrashed = $thread->trashed();
+
+        if ($this->isPermaDeleting() && method_exists($thread, 'forceDelete'))
         {
+            $thread->posts()->withTrashed()->forceDelete();
             $thread->forceDelete();
         }
         else
         {
             $thread->delete();
         }
 
-        $thread->category->syncCurrentThreads();
+        $category = $thread->category;
+
+        if (! $threadIsTrashed)
+        {
+            // Only change category stats and FKs if the thread wasn't already soft-deleted
+            $category->update([
+                'newest_thread_id' => $category->getNewestThreadId(),
+                'latest_active_thread_id' => $category->getLatestActiveThreadId(),
+                'thread_count' => DB::raw('thread_count - 1'),
+                'post_count' => DB::raw("post_count - {$thread->postCount}")
+            ]);
+        }
 
         return $thread;
     }
+
+    private function isPermaDeleting(): bool
+    {
+        return ! config('forum.general.soft_deletes') || isset($this->validated()['permadelete']) && $this->validated()['permadelete'];
```

Comment:
`isset($this->validated()['permadelete']) && $this->validated()['permadelete'];` get used a lot in one form or another. Could be an idea to refactor all these calls into a helper function of some kind - doesn't need doing now though.